vmdb/spec/requests/api/vms_spec.rb

```diff
@@ -23,1008 +33,764 @@ def app
   end
 
   context "Vm accounts subcollection" do
-    it "query VM accounts subcollection with no related accounts" do
-      basic_authorize @cfme[:user], @cfme[:password]
-
-      vm = FactoryGirl.create(:vm_vmware)
-
-      @success = run_get "#{@cfme[:vms_url]}/#{vm.id}/accounts"
+    let(:acct1) { FactoryGirl.create(:account, :vm_or_template_id => vm.id, :name => "John") }
+    let(:acct2) { FactoryGirl.create(:account, :vm_or_template_id => vm.id, :name => "Jane") }
+    let(:vm_accounts_url)      { "#{vm_url(vm.id)}/accounts" }
+    let(:acct1_url)            { "#{vm_accounts_url}/#{acct1.id}" }
+    let(:acct2_url)            { "#{vm_accounts_url}/#{acct2.id}" }
+    let(:vm_accounts_url_list) { [acct1_url, acct2_url] }
+
+    context "query VM accounts subcollection with no related accounts" do
+      before do
+        api_basic_authorize
+
+        run_get vm_accounts_url
+      end
 
-      expect(@code).to eq(200)
-      expect(@result).to have_key("name")
-      expect(@result["name"]).to eq("accounts")
-      expect(@result["resources"]).to be_empty
+      include_examples "empty_query_result", :accounts
     end
 
-    it "query VM accounts subcollection with two related accounts" do
-      basic_authorize @cfme[:user], @cfme[:password]
-
-      vm = FactoryGirl.create(:vm_vmware)
-      acct1 = FactoryGirl.create(:account, :vm_or_template_id => vm.id, :name => "John")
-      acct2 = FactoryGirl.create(:account, :vm_or_template_id => vm.id, :name => "Jane")
+    context "query VM accounts subcollection with two related accounts" do
+      before do
+        api_basic_authorize
+        acct1
+        acct2
```

Comment:
This is the usecase for `let!(:resource) { create ... }`
Since these `let`s are common across the tests, no reason to go there. `let` works fine.

I wish it weren't necessary, but it works and makes sense. Please don't add no op code like `create_resources`. The developer will track down the method only to find it empty and baffling. It looks like it could be optimized away.

I'd just put a comment in the code and be done with it: `# ensure objects created in the database`.

If you do find resources that need to be loaded across all tests, maybe `let!(:resource)` works for you